cmd/kubeadm/app/phases/addons/dns/dns.go

```diff
@@ -297,6 +298,83 @@ func createDNSService(dnsService *v1.Service, serviceBytes []byte, client client
 	return nil
 }
 
+func createCoreDNSConfigMap(client clientset.Interface, cm *v1.ConfigMap) error {
+	// Get the CoreDNS ConfigMap. If not present, Create a new default ConfigMap.
+	if _, err := client.CoreV1().ConfigMaps(cm.ObjectMeta.Namespace).Get(kubeadmconstants.CoreDNSConfigMap, metav1.GetOptions{}); err != nil {
```

Comment:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/util/apiclient/idempotency.go#L96
can `CreateOrRetainConfigMap` be reused here?